skada/deep/utils.py

```diff
@@ -52,3 +55,33 @@ def check_generator(seed):
     raise ValueError(
         "%r cannot be used to seed a torch.Generator instance" % seed
     )
+
+
+def _infer_predict_nonlinearity(net):
+    """Infers the correct nonlinearity to apply for this net
+
+    The nonlinearity is applied only when calling
+    :func:`~skorch.classifier.NeuralNetClassifier.predict` or
+    :func:`~skorch.classifier.NeuralNetClassifier.predict_proba`.
+
+    """
+    # Implementation: At the moment, this function "dispatches" only
+    # based on the criterion, not the class of the net. We still pass
+    # the whole net as input in case we want to modify this at a
+    # future point in time.
+    if len(net._criteria) != 1:
+        # don't know which criterion to consider, don't try to guess
+        return _identity
+
+    criterion = getattr(net, net._criteria[0] + '_').base_criterion
+
+    if isinstance(criterion, CrossEntropyLoss):
+        return partial(torch.softmax, dim=-1)
+
+    if isinstance(criterion, BCEWithLogitsLoss):
+        return _sigmoid_then_2d
+
+    if isinstance(criterion, BCELoss):
+        return _make_2d_probs
```

Comment:
Here shouldnt be inverted? The `BCEWithLogitsLoss` "combines a Sigmoid layer and the BCELoss in one single class".
Thus having `_sigmoid_then_2d` for `BCELoss` and `_make_2d_probs` for `BCEWithLogitsLoss`.

Reply:
I copied/pasted the skorch function

Reply:
But yeah it's true. I think they made a mistake